core/domain/user_jobs_one_off.py

```diff
@@ -360,6 +364,130 @@ def map(user_model):
         user_model.put()
 
 
+class DraftChangesMathRichTextInfoModelGenerationOneOffJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """Job that finds all the valid exploration draft changes with math rich
+    text components and creates a temporary storage model with all the
+    information required for generating math rich text component SVG images.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [user_models.ExplorationUserDataModel]
+
+    @staticmethod
+    def map(item):
+        exp_id = item.exploration_id
+        exploration = exp_fetchers.get_exploration_by_id(exp_id)
+        draft_change_list = [
+            exp_domain.ExplorationChange(change)
+            for change in item.draft_change_list]
+        draft_change_list_version = item.draft_change_list_exp_version
+        exploration_version = exploration.version
+        final_draft_change_list = None
+
+        if exploration_version == draft_change_list_version:
+            final_draft_change_list = draft_change_list
+        elif exploration_version > draft_change_list_version:
+            updated_draft_change_list = (
+                draft_upgrade_services.try_upgrading_draft_to_exp_version(
+                    draft_change_list, draft_change_list_version,
+                    exploration_version, exp_id))
+            final_draft_change_list = updated_draft_change_list
+
+
+        if final_draft_change_list is not None:
+            try:
+                html_string = ''.join(
+                    draft_upgrade_services.
+                    extract_html_from_draft_change_list(
+                        final_draft_change_list))
+                latex_values = (
+                    html_validation_service.
+                    extract_latex_values_from_math_rich_text_without_filename(
+                        html_string))
+                if len(latex_values) > 0:
+                    math_rich_text_info = (
+                        exp_domain.ExplorationMathRichTextInfo(
+                            latex_values))
+                    approx_size_of_math_svgs_bytes = (
+                        math_rich_text_info.get_svg_size_in_bytes())
```

Comment:
Follow the comments in exp_one_off_job.py file over here as well*

Reply:
changed this